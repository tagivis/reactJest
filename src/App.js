import React from 'react';
import logo from './logo.svg';
import './App.css';
import Card from './Card/Card';
import Counter from './Counter/Counter';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <Card/>
        <Counter/>
      </header>
    </div>
  );
}

export default App;
