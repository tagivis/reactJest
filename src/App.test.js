import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

import{sum} from './math'

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('Validar una suma', ()=>{
  it('suma dos numeros', ()=>{
    expect(sum(1,2)).toEqual(3);
    expect(sum(2,2)).toEqual(4);
  })
});


test('el componente se muestra', () => {
  const component = renderer.create(<App/>);
  let tree= component.toJSON();

  expect(tree).toMatchSnapshot();

});



