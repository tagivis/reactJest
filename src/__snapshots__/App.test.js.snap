// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`el componente se muestra 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div>
      <img
        alt="face"
        src="https://source.unsplash.com/random/250x250/?face"
      />
      <h2>
        Alex
      </h2>
      <h3>
        25
      </h3>
    </div>
    <button
      className="suma"
      onClick={[Function]}
    >
      Add one
    </button>
    <button
      className="resta"
      onClick={[Function]}
    >
      Minus one
    </button>
    <p>
      La cuenta es 
      0
    </p>
  </header>
</div>
`;
